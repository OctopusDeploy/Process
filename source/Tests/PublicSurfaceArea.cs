using System.Linq;
using System.Reflection;
using Assent;
using NUnit.Framework;
using Octopus.SilentProcessRunner;

namespace Tests
{
    public class PublicSurfaceArea
    {
        [Test]
        public void TheLibraryOnlyExposesWhatWeWantItToExpose()
        {
            var assembly = typeof(SilentProcessRunner).Assembly;
            var publicMembers =
                from t in assembly.GetExportedTypes()
                from m in t.GetMembers(BindingFlags.Public | BindingFlags.DeclaredOnly | BindingFlags.Static | BindingFlags.Instance)
                where !(m is MethodBase method && method.IsSpecialName)
                select $"{m.DeclaringType?.FullName}.{m.Name}";

            this.Assent(string.Join("\r\n", publicMembers));
        }
    }
}