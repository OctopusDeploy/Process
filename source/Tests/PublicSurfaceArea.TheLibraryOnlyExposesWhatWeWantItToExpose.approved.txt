Octopus.Shellfish.ShellCommand.WithWorkingDirectory
Octopus.Shellfish.ShellCommand.WithArguments
Octopus.Shellfish.ShellCommand.BeforeStartHook
Octopus.Shellfish.ShellCommand.WithRawArguments
Octopus.Shellfish.ShellCommand.WithEnvironmentVariables
Octopus.Shellfish.ShellCommand.RunAsUser
Octopus.Shellfish.ShellCommand.WithOutputEncoding
Octopus.Shellfish.ShellCommand.CaptureStdOutTo
Octopus.Shellfish.ShellCommand.CaptureStdOutTo
Octopus.Shellfish.ShellCommand.CaptureStdErrTo
Octopus.Shellfish.ShellCommand.CaptureStdErrTo
Octopus.Shellfish.ShellCommand.KillProcessOnCancellation
Octopus.Shellfish.ShellCommand.SwallowCancellationException
Octopus.Shellfish.ShellCommand.Execute
Octopus.Shellfish.ShellCommand.ExecuteAsync
Octopus.Shellfish.ShellCommandExtensionMethods.ExecutableCanBeDotnetDll
Octopus.Shellfish.ShellCommandResult.ExitCode
Octopus.Shellfish.ShellExecutionException.Errors
Octopus.Shellfish.ShellExecutionException.Message
Octopus.Shellfish.ShellExecutor.ExecuteCommand
Octopus.Shellfish.ShellExecutor.ExecuteCommandWithoutWaiting